src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/BuildIntegratedProjectAction.cs

```diff
@@ -57,6 +64,7 @@ public class BuildIntegratedProjectAction : NuGetProjectAction
         {
         }
 
+        [Obsolete("This type is not expected to be created externally.")]
```

Comment:
Just to clarify, this [constructor](https://learn.microsoft.com/dotnet/api/system.obsoleteattribute.-ctor?view=net-7.0#system-obsoleteattribute-ctor(system-string)) triggers a warning instead of an error. If we want to raise an error, then we need to use a different constructor.
```cs
[Obsolete("This type is not expected to be created externally.", error: true)]
```

Reply:
that would make it a breaking change. I mean, warnings are too if customers use treat warnings as errors, but at least in that case they're shotting themselves in the foot. There have been customers asking us questions about the NuGet.PackageManagement package before, so we know that some people are using it to install packages in their own apps.

Reply:
I agree with @zivkan. 

The error option is a really big hammer. We haven't really gotten a lot of feedback on our deprecations to change our approach imo.